### ISSUE TITLE
Seminar Talk: Co-learning Planning and Control Policies Using Differentiable Formal Task Constraints

### ISSUE THREAD

**Issue by ZikangXiong** [PurduePL/purduepl.github.io#109]
### Title
Co-learning Planning and Control Policies Using Differentiable Formal Task Constraints

### Abstract
Our paper presents a hierarchical reinforcement learning algorithm constrained by differentiable signal temporal logic. Previous work on logic-constrained reinforcement learning consider encoding these constraints with a reward function, constraining policy updates with a sample-based policy gradient. However, such techniques oftentimes tend to be inefficient because of the significant number of samples required to obtain accurate policy gradients. In this paper, instead of implicitly constraining policy search with sample-based policy gradients, we directly constrain policy search by backpropagating through formal constraints, enabling training hierarchical policies with substantially fewer training samples. The use of hierarchical policies is recognized as a crucial component of reinforcement learning with task constraints. We show that we can stably constrain policy updates, thus enabling different levels of the policy to be learned simultaneously, yielding superior performance compared with training them separately. Experiment results on several simulated high-dimensional robot dynamics and a real-world differential drive robot (TurtleBot3) demonstrate the effectiveness of our approach on five different types of task constraints. Demo videos, code, and models can be found at our project website: https://sites.google.com/view/dscrl

### Bio
I am currently pursuing a Ph.D. in Computer Science at Purdue University, where I am working under the guidance of Suresh Jagannathan. My research lies in the intersection of control theory, deep reinforcement learning, and neuro-symbolic reasoning. My research goal is to provide verifiable, robust, and programmable learning frameworks for AI and robotics systems.
